rust/agama-cli/src/lib.rs

```diff
@@ -198,6 +202,22 @@ async fn allowed_insecure_api(use_insecure: bool, api_url: String) -> Result<boo
     }
 }
 
+pub fn download_file(url: &str, path: &PathBuf) -> Result<(), ServiceError> {
+    let mut file = fs::OpenOptions::new()
+        .create(true)
+        .truncate(true)
+        .write(true)
+        .mode(0o400)
+        .open(path)
+        .context(format!("Cannot write the file '{}'", path.display()))?;
+
+    match Transfer::get(&url, &mut file) {
+        Ok(()) => println!("File saved to {}", path.display()),
```

Comment:
we probably should introduce in future some quiet and verbose options.

Reply:
Yes, in the future we should think about that. The CLI needs more love.